### PR TITLE
QUARKUS-2387 - Promote Mailer extension from TP to full support

### PR DESCRIPTION
**Pull request by rsvoboda**
QUARKUS-2387 - Promote Mailer extension from TP to full support

https://issues.redhat.com/browse/QUARKUS-2387

### CODE REVIEW

File: QUARKUS-2387.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# QUARKUS-2387 - Promote Mailer extension from TP to full support
+
+Evaluate the extension for promotion to full support. Propose needed enhancements and report found bugs. 
+
+## Scope of the testing
+ - Review of test coverage in Quarkus integration tests
+ - Review upstream guides for the extension
+ - Exploratory testing of the extension
```
Comment (rsvoboda) [inline QUARKUS-2387.md:8]: I should probably stress out that I will use gmail config with the exploratory testing

Comment (rsvoboda) [reply QUARKUS-2387.md:8]: done

**Comment by rsvoboda** [comment]
Added note about QS

**Review by michalvavrik — APPROVED** [review]
Hello @rsvoboda ,

lgtm, I want to suggest couple of topics that should be covered by us anyway as they are _must do_ for any serious app and AFAIK upstream is using mocks:
- a source address, destination addresses (.., -> email header encoding) and message data encoding
  - Email format, make sure: 
    - for compatibility reasons, e-mail address is encoded with the 7-bit ASCII as defined by RFC 2821
    - for non-ascii chars (very common, trust me :-( ), you can use Punycode encoding (see RFC 3492) for local part of the address
    - email headers use MIME encoded-word syntax
  - Specify headers, e.g. for message body, you can specify `Content-Transfer-Encoding` (e.g. base64, 8bit stuff, quoted-printable encoding,...) and than encode body section of the email accordingly. For AWS, you can use headers to specify SES configuration etc. Important.
- file attachments - examples shows you can specify content type, is it possible to specify name too? e.g. `Content-Type: image/jpeg; name="leo-messi.jpeg"`, usually, you have got a file (e.g. stored on cloud or whatever) and display name differs from actual name of the file (name often contains username, date, source whatever)

... I've couple more on mind, but I'd need time to write it down

File: QUARKUS-2387.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# QUARKUS-2387 - Promote Mailer extension from TP to full support
+
+Evaluate the extension for promotion to full support. Propose needed enhancements and report found bugs. 
+
+## Scope of the testing
+ - Review of test coverage in Quarkus integration tests
+ - Review Quarkus Quickstart for the extension
+ - Review upstream guides for the extension
+ - Exploratory testing of the extension
+   - Using Gmail configuration
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply QUARKUS-2387.md:10]: I will try that, I can put it into TP directly.

Was also considering experiments with Sendgrid

**Comment by rsvoboda** [comment]
@michalvavrik PR updated to be more explicit about exploratory testing and scope of the extension

**Review by michalvavrik — APPROVED** [review]
I like the changes, thank you

**Pull request merged**